src/main/java/animals/service/FileService.java

```diff
@@ -0,0 +1,45 @@
+package animals.service;
+
+import animals.model.Node;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+
+public abstract class FileService {
+
+    public void save(Node root) {
+        String fileName = getFileName();
+        ObjectMapper objectMapper = getObjectMapper();
+        try {
+            objectMapper
+                    .writerWithDefaultPrettyPrinter()
+                    .writeValue(new File(fileName), root);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Node load() {
+        String fileName = getFileName();
+        Node root = new Node();
+        ObjectMapper objectMapper = getObjectMapper();
```

Comment:
objectMapper is used only inside try/catch. It is better to move objectMapper inside try/catch.
We should declare variables close to place where it is used.
Please check other places as well